python/fbprophet/forecaster.py

```diff
@@ -1037,6 +1063,11 @@ def predict(self, df=None):
                 raise ValueError('Dataframe has no rows.')
             df = self.setup_dataframe(df.copy())
 
+        if self.calculate_holidays == True:
```

Comment:
As described in the comment above, if this is done inside `make_holiday_features` then it doesn't have to be in both fit and predict.